lib/kennel.rb

```diff
@@ -67,11 +67,14 @@ def store(parts)
       Progress.progress "Storing" do
         old = Dir["generated/{#{(project_filter || ["**"]).join(",")}}/*"]
         used = []
+        mutex = Mutex.new
 
         Utils.parallel(parts, max: 2) do |part|
           path = "generated/#{part.tracking_id.tr("/", ":").sub(":", "/")}.json"
-          used << File.dirname(path) # only 1 level of sub folders, so this is safe
-          used << path
+          mutex.synchronize do
+            used << File.dirname(path) # only 1 level of sub folders, so this is safe
+            used << path
+          end
```

Comment:
prefer concat which is thread-safe
```suggestion
          used.concat [File.dirname(path), path] # only 1 level of sub folders, so this is safe (use concat for thread-safety)
```